doc/source/io.rst

```diff
@@ -440,6 +440,64 @@ individual columns:
     Specifying ``dtype`` with ``engine`` other than 'c' raises a
     ``ValueError``.
 
+.. note::
+
+   Reading in data with columns containing mixed dtypes and relying
+   on ``pandas`` to infer them is not recommended. In doing so, the
+   parsing engine will infer the dtypes for different chunks of the data,
+   rather than the whole dataset at once. Consequently, you can end up with
+   column(s) with mixed dtypes. For example,
+
+   .. ipython:: python
+         :okwarning:
+
+       df = pd.DataFrame({'col_1':range(500000) + ['a', 'b'] + range(500000)})
+       df.to_csv('foo')
+       mixed_df = pd.read_csv('foo')
+       mixed_df['col_1'].apply(type).value_counts()
+       mixed_df['col_1'].dtype
+
+   will result with `mixed_df` containing an ``int`` dtype for certain chunks
```

Comment:
> due to a problem during parsing.

explain what you actually mean here. There wasn't a problem, but the data instead had mixed dtypes.